### PR TITLE
Add aws-k8s-1.21, vmware-k8s-1.21 variants

### LINKED ISSUE #1611: Kubernetes 1.21 support
Kubernetes 1.21 was released in April 2021: https://kubernetes.io/blog/2021/04/08/kubernetes-1-21-release-announcement/

[CHANGELOG since 1.20.0](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md#changelog-since-v1200)

### PR DESCRIPTION
**Pull request by etungsten**
**Issue number:**
https://github.com/bottlerocket-os/bottlerocket/issues/1611

**Description of changes:**

For reference, the addition of aws-k8s-1.20 was in https://github.com/bottlerocket-os/bottlerocket/pull/1437.

Version of `kubelet` packaged: v1.21.2

```
Author: Erikson Tung <etung@amazon.com>
Date:   Tue Jun 15 11:31:57 2021 -0700

    Add aws-k8s-1.21 variant
    
    Adds aws-k8s-1.21 variant, relinks symlinks in models

```
```
Author: Erikson Tung <etung@amazon.com>
Date:   Tue Jun 15 11:55:19 2021 -0700

    Add vmware-k8s-1.21 variant
    
    Adds vmware-k8s-1.21 variant, relinks symlinks in models

```

```
Author: Erikson Tung <etung@amazon.com>
Date:   Tue Jun 15 11:30:21 2021 -0700

    packages: add kubernetes-1.21
    
    * No longer need '0001-always-set-relevant-variables-for-cross-compiling.patch'
    * Third party package 'github.com/miekg/dns' version bump included some
    changes in its license file so the hash changed. The license is still
    'BSD-3-Clause'.
```

`0001-always-set-relevant-variables-for-cross-compiling.patch` is no longer needed as a result of the following changes in 1.21:
* https://github.com/kubernetes/kubernetes/commit/fd01467c0c3211f7014e7bf3c6fa62ef87d65240 
* https://github.com/kubernetes/kubernetes/commit/f19f22b43edf498444817a0052b71d6084e68db9
We export the `KUBE_*_CC` variables directly.

**Testing done:**
- [x] Pods run OK

- [x]  Conformance testing with 1.21 control plane and x86_64 aws-k8s-1.21 nodes

- [x]  Conformance testing with 1.21 control plane and aarch64 aws-k8s-1.21 nodes

- [x]  Conformance testing with 1.21 control plane and x86_64 vmware-k8s-1.21 nodes

- [x] Confirm the EBS CSI driver still works.

- [x] Confirm the vSphere CSI driver still works - (Block volumes only since file volumes are not yet supported on AWS VMC)

- [x] Check kubelet logs and system logs in general, compared them with the aws-k8s-1.20 variant kubelet logs:

There's a new kubelet warning during start up:
```
Flag --network-plugin has been deprecated, will be removed along with dockershim.
```
dockershim for kubelet will be fully deprecated by 1.23 which I believe we'll have to address separately at some point.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: packages/kubernetes-1.21/kubernetes-1.21.spec
Comment (tjkirch) [inline packages/kubernetes-1.21/kubernetes-1.21.spec:15]: This snippet could definitely use a comment - something about how the k8s build process uses this to determine the cross-compiler for a target architecture, maybe with an example key=value?

File: packages/kubernetes-1.21/kubelet.service
Comment (tjkirch) [inline packages/kubernetes-1.21/kubelet.service:1]: Do kubelet logs look different?  Is there an additional log file?  https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md#structured-logging-in-kubelet

**Review by tjkirch — COMMENTED** [review]
LGTM pending testing.

File: packages/kubernetes-1.21/kubelet.service
Comment (etungsten) [reply packages/kubernetes-1.21/kubelet.service:1]: The logs do not look different from k8s-1.20. Not sure if this is something we have to enable.

**Comment by etungsten** [comment]
Push above separates the commits into 3, first one adds the kubernetes-1.21 package, one that adds the aws-k8s-1.21 variant, and one that adds the vmware-k8s-variant.

**Comment by etungsten** [comment]
Push above adds some clarifying comments about `kube_cc` in `kubernetes-1.21.spec`

File: README.md
Comment (tjkirch) [inline README.md]:
```suggestion
Other variants we have in preview are designed to be Kubernetes worker nodes in VMware:
```

File: variants/vmware-k8s-1.21/Cargo.toml
Comment (tjkirch) [inline variants/vmware-k8s-1.21/Cargo.toml:15]: Looks like you started before the kdump support was merged, we'll want to add that.

File: variants/vmware-k8s-1.21/Cargo.toml
Comment (tjkirch) [inline variants/vmware-k8s-1.21/Cargo.toml:1]: If we're adding VMware, we should also update QUICKSTART-VMWARE.md, at least for grammar around multiple VMware variants, but possibly also to make 1.21 the new de facto default.

**Comment by etungsten** [comment]
Push above rebases onto develop

**Comment by etungsten** [comment]
Push above addresses comments from @tjkirch

**Comment by etungsten** [comment]
Updated PR description to include testing for vsphere conformance testing, vsphere csi driver testing against a 1.21 cluster control plane.

**Review by tjkirch — APPROVED** [review]

**Comment by etungsten** [comment]
Github actions is consistently failing checks. I added three more jobs as part of this PR. There are now a total of 18 jobs to run per PR. It seems like one or two of the jobs always times out for some reason.

**Review by zmrow — APPROVED** [review]
:money_with_wings:

**Comment by etungsten** [comment]
[v1.21.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.2) just got released. I'll update the packaged kubelet version soon.

**Review by tjkirch — APPROVED** [review]
LGTM, assuming 1.21.2 was retested.

**Review by webern — APPROVED** [review]

**Comment by etungsten** [comment]
Push above updates kubelet to v1.21.2. Pods run fine.

Build checks keep failing due to network timeouts etc. Failures are not related to the changes.

**Comment by etungsten** [comment]
Rebases onto develop to bring in the commit that fixes CI.

**Review by tjkirch — APPROVED** [review]

**Review by bcressey — APPROVED** [review]

**Pull request merged**
